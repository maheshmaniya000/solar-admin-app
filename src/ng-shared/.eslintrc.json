{
  "extends": ["../../.eslintrc-angular.json", "../../.eslintrc.json"],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts"],
      "parserOptions": {
        "project": [
          "src/ng-shared/tsconfig.(lib|spec).json",
          "src/ng-shared/.storybook/tsconfig.json"
        ]
      },
      "rules":{
        "max-classes-per-file": "warn" // TODO switch to 'error' / remove when all resolved
      }
    },
    {
      "files": ["*.stories.ts"],
      "rules": {
        "max-classes-per-file": "off"
      }
    }
  ]
}
