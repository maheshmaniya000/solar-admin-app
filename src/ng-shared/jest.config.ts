import { createAngularJestConfig } from '../../tools/jest-config-factory';

export default createAngularJestConfig({
  name: 'shared',
  library: true
});
