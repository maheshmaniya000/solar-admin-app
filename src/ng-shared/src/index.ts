export * from './lib/user/types';
export * from './lib/config';
export * from './lib/utils/index';
