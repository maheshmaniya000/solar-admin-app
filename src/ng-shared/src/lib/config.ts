import { Injectable } from '@angular/core';

import { DataLayerDef } from '@solargis/dataset-core';
import { MapLayerDef, MapSelectedLayers } from '@solargis/types/map';
import { SolargisApp } from '@solargis/types/user-company';

export type AmplitudeAnalyticsConfig = {
  key: string;
};

export type LambdaApiConfig = {
  baseUrl: string;
  customerUrl: string;
  projectUrl: string;
  prospectUrl: string;
  userTagUrl: string;
};

export type AppConfig = { url: string; devUrl?: string };

export type Auth0Config = {
  customNamespace: string;
  passwordMatchRegexp: string;
};

export type GoogleAnalyticsConfig = {
  linkerAutolink?: string[];
  trackingId: string;
};

export type GeolocatorConfig = {
  enabled: boolean;
  fallback: string;
};

export type ApiConfig = {
  key: string;
  url: string;
};

export type RecaptchaConfig = {
  publicKey: string;
};

export type SentryConfig = {
  dsn: string;
  release: string;
};

// Maps to @solargis/ng-geosearch#GeocoderConfig
// Only geocoder API keys -> we can turn off/on geocoders with .env
export type GeocoderKeysConfig = {
  nominatim?: boolean;
  opencage?: string;
};

export type DataConfig = {
  defaultLayerKey: string;
  ltaLayers: DataLayerDef[];
  pvcalcLayers: DataLayerDef[];
};

export type MapConfig = {
  center: string;
  layers: MapLayerDef[];
  selected: MapSelectedLayers;
};

export type ReportConfig = {
  assetsUrl: string;
};

// http://plnkr.co/edit/87k3AQbBZXV5c7lcKFRS

@Injectable()
export class Config {
  amplitude?: AmplitudeAnalyticsConfig;
  api: LambdaApiConfig;
  apps: { [app: string]: AppConfig };
  auth0: Auth0Config;
  env: string;
  ga?: GoogleAnalyticsConfig;
  geolocator: GeolocatorConfig;
  ip?: string;
  ipstack: ApiConfig;
  languages: string[];
  recaptcha: RecaptchaConfig;
  sentry?: SentryConfig;
  geocoder?: GeocoderKeysConfig;
  version: string;
  sdatDownloadUrl?: string;
}

@Injectable()
export class AdminAppConfig extends Config {

}

export interface HasMapAndDataConfig {
  data: DataConfig;
  map: MapConfig;
}

export interface HasReportConfig {
  report: ReportConfig;
}

@Injectable()
export class ProspectAppConfig extends Config implements HasMapAndDataConfig, HasReportConfig {
  data: DataConfig;
  map: MapConfig;
  report: ReportConfig;
  timezonedb: ApiConfig;
}

@Injectable()
export class EvaluateAppConfig extends Config {
  // TODO add evaluate specific config
  // TOTO refactor shared config between prospect and evaluate, e.g. map, timezonedb
}

@Injectable()
export class DashboardAppConfig extends Config {
  // TODO add dashboard specific config
  map: MapConfig;
}

@Injectable()
export class PrintAppConfig extends Config implements HasReportConfig {
  report: ReportConfig;
}

// https://github.com/angular/angular/issues/23279#issuecomment-539435523

export async function fetchConfig(
  app: 'admin' | 'print' | 'dashboard' | SolargisApp
): Promise<Config> {
  const res = await fetch(`/api/config/${app}`);
  return (await res.json()) as Config;
}
