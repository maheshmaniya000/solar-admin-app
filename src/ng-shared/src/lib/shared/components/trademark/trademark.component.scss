
@import 'theme';

span {
  font-weight: 400;
  color: map-get($primary, 800);
  font-size: $sg-12;
}
