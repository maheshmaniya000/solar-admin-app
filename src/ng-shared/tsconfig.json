{
  "extends": "../../tsconfig.base.json",
  "files": [],
  "references": [{ "path": "./tsconfig.lib.json" }, { "path": "./tsconfig.spec.json" }, { "path": "./.storybook/tsconfig.json" }],
  "compilerOptions": {
    "strict": false, // FIXME
    "noImplicitReturns": false // FIXME
  },
  "angularCompilerOptions": {
    "strictInjectionParameters": false, // FIXME
    "strictTemplates": false // FIXME
  }
}
