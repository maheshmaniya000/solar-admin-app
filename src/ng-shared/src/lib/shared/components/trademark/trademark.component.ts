import { Component } from '@angular/core';

import { Config } from 'ng-shared/config';

@Component({
  selector: 'sg-trademark',
  template: `<span>Â© {{ year }} Solargis v {{ config.version }}</span>`,
  styleUrls: ['./trademark.component.scss']
})
export class TrademarkComponent {
  year = new Date().getFullYear();

  constructor(readonly config: Config) {}
}
