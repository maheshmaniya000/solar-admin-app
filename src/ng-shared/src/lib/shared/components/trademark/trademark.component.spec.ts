import { createComponentFactory, mockProvider, Spectator } from '@ngneat/spectator/jest';

import { Config } from 'ng-shared/config';

import { TrademarkComponent } from './trademark.component';

describe('TrademarkComponent', () => {
  describe('component', () => {
    const createComponent = createComponentFactory({
      component: TrademarkComponent,
      providers: [
        mockProvider(Config, {
          version: '1.2.3'
        })
      ],
      shallow: true
    });
    let spectator: Spectator<TrademarkComponent>;

    beforeEach(() => {
      jest
        .useFakeTimers('modern')
        .setSystemTime(new Date(2020, 6));
      spectator = createComponent();
      jest.useRealTimers();
    });

    it(`should correctly render component for year 2020 and version 1.2.3`, () => {
      expect(spectator.fixture).toMatchSnapshot();
    });
  });
});
