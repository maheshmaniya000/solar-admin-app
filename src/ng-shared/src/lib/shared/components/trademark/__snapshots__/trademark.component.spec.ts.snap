// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrademarkComponent component should correctly render component for year 2020 and version 1.2.3 1`] = `
<sg-trademark
  __ngContext__={[Function LRootView]}
  config={[Function Object]}
  year={[Function Number]}
>
  <span>
    Â© 2020 Solargis v 1.2.3
  </span>
</sg-trademark>
`;
